### ISSUE TITLE
[BUG] Error NETSDK1005 after changes in .csproj file

### ISSUE THREAD

**Issue by tsjdev-apps** [dotnet/maui#6626]
### Description

I've created a new .NET MAUI app using the latest RC1 version. I'm opening the project file of the project and do some changes like chaing the `ApplicationTitle` or the `ApplicationDisplayVersion`. If I'm now also updating the `Color` of the `MauiSplashScreen` the app won't build with the error `NETSDK1005`. Also undo the latest changes won't fix the issue.

### Steps to Reproduce

1. Create a new .NET MAUI app
2. Edit the project file
3. Update `ApplicationTitle`,  `ApplicationDisplayVersion`, `Color` of `MauiSplashScreen`

### Version with bug

Release Candidate 1

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Android 11, Windows 10.0.19041

### Did you find any workaround?

-

### Relevant log output

```shell
error NETSDK1005: Assets file '\obj\project.assets.json' doesn't have a target for 'net6.0-windows10.0.19041'. Ensure that restore has run and that you have included 'net6.0-windows10.0.19041' in the TargetFrameworks for your project.
```

**Comment by rasert** [comment]
Is there a workaround for this?

**Comment by jonathanpeppers** [comment]
@rasert set `$(Version)` and `$(ApplicationDisplayVersion)` to the same value in your project?

**Comment by KiristonK** [comment]
Hi, on Visual Studio Community 17.2.0 Preview 6.0 still having this issue.
![obraz](https://user-images.githubusercontent.com/48825244/167072296-3b74d53d-465f-41c5-9ef6-62dc4c929464.png)

After changing application display version to anything different from 1.0 or 1.0.0 i'm getting this error:
`Severity	Code	Description	Project	File	Line	Suppression State
Error	NETSDK1005	Assets file '_PROJECT_PATH_\obj\project.assets.json' doesn't have a target for 'net6.0-android'. Ensure that restore has run and that you have included 'net6.0-android' in the TargetFrameworks for your project.	_PROJECT_NAME_	C:\Program Files\dotnet\sdk\6.0.300-preview.22204.3\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets	267	
`

And contents of project.assets.json:
![obraz](https://user-images.githubusercontent.com/48825244/167072372-e5bd6ad5-6641-4373-a99a-1135ef957e96.png)

The "logs" section in project.assets.json:
`"logs": [
    {
      "code": "NU1105",
      "level": "Error",
      "message": "Unable to read project information for '_PROJECT_NAME_	': Sequence contains more than one element"
    }
  ]`

Is there any way to set version higher than 1.0.0?

**Comment by jonathanpeppers** [comment]
@KiristonK did you already try setting `$(ApplicationDisplayVersion)` and `$(Version)` to the same value? I don't see that in your screenshots.

The fix for this is not released yet.

**Comment by KiristonK** [comment]
@jonathanpeppers Last time i've tried this, nothing chaned, but two days later everything with $(Version) and $(ApplicationDisplayVersion) set to the same value - it works just fine...
Anyway, thanks for help :)

**Comment by pauldendulk** [comment]
I still have this problem today with the GA workloads and VS 17.3.0 Preview 1.1 (that is the latest update). My problem was resolved by deleting ApplicationDisplayVersion from the csproj but it took me quite some time to find this workaround. 

In my situation any new MAUI project added to an existing solution (with a lot of older projects) did not compile. No problem with a newly created solution.

**Comment by jonathanpeppers** [comment]
@pauldendulk can you install [Project System Tools](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProjectSystemTools2022) and record a `.binlog` for the design-time build?

`View > Other Windows > Build Logging` and hit the play button. Hopefully you can get a design-time build to appear that fails and share it with us? Thanks!

If it happens in one project but not the others, you might be hitting the general issue in NuGet: https://github.com/NuGet/Home/issues/6461

**Comment by Redth** [comment]
Reopened as I'm seeing this again in .NET 8 P4 with 17.7P1.

It looks like the logic to set <Version> needs to shift back into the workload so it happens soon enough.

**Comment by Redth** [comment]
For now the easiest workaround is to add `<Version>$(ApplicationVersion)</Version>` in your project directly after your `<ApplicationVersion>1.0</ApplicationVersion>` property.

**Comment by mattleibow** [comment]
I believe that the issue you are seeing @Redth is https://github.com/dotnet/maui/issues/10579

This particular one was a result for the TFMs being passed as windows to all dependencies. I moved the code we added back into the workload but nothing changed. And looking at what we have, the code that sets the property is only imported using:

```xml
<Import Project="WinUI.targets" Condition=" '$(TargetPlatformIdentifier)' == 'windows' and '$(WindowsAppSDKWinUI)' == 'true'" />
```

This had a condition for the Windows TFM and the `WindowsAppSDKWinUI` property - which is only set by the Windows App SDK NuGet - after the install has completed. So this never ran on before the restore.

I tested this with the latest stable .NET MAUI - 7.0.86 - and I was able to reproduce this by simply updating the 2 properties (only the first matters really):

```xml
<ApplicationDisplayVersion>2.0</ApplicationDisplayVersion>
<ApplicationVersion>2</ApplicationVersion>
```
